driver.sh

```diff
@@ -139,14 +140,51 @@ setup_variables() {
 }
 
 check_dependencies() {
+  # Check for existence of needed binaries
   command -v nproc
   command -v "${CROSS_COMPILE:-}"as
   command -v ${qemu}
   command -v timeout
   command -v unbuffer
-  command -v clang-9
-  command -v llvm-ar-9
-  command -v "${LD:="${CROSS_COMPILE:-}"ld}"
+
+  # Check for LD, CC, and AR environmental variables
+  # and print the version string of each. If CC and AR
+  # don't exist, try to find them.
+  # lld isn't ready for all architectures so it's just
+  # simpler to fall back to GNU ld when LD isn't specified
+  # to avoid architecture specific selection logic.
+
+  "${LD:="${CROSS_COMPILE:-}"ld}" --version
+
+  if [[ -z "${CC:-}" ]]; then
+    for CC in clang-9 clang-8 clang-7 clang; do
+      ${CC} --version 2>/dev/null && break
+    done
+  else
+    ${CC} --version
+  fi
+
+  if [[ -z "${AR:-}" ]]; then
+    for AR in llvm-ar-9 llvm-ar-8 llvm-ar-7 llvm-ar "${CROSS_COMPILE:-}"ar; do
+      ${AR} --version 2>/dev/null && break
+    done
+    # Optimistically check to see that the user has a llvm-ar
+    # with https://reviews.llvm.org/rL354044. If they don't,
+    # fall back to GNU ar and let them know.
+    if ${AR} --version | grep -q "LLVM" && \
+       [[ $(${AR} --version | grep version | sed -e 's/.*LLVM version //g' -e 's/[[:blank:]]*$//' -e 's/\.//g') -lt 900 ]]; then
+      set +x
```

Comment:
Can you put this check in another function with a descriptive name like `check_ar_version` or w/e?

Reply:
Yes.